### PR TITLE
Updated to new quilt commands and changed CPU throttling

### PR DESCRIPTION
**Pull request by atalreja**

### CODE REVIEW

File: src/measure_performance_MEAN_py3.py [hunk 1]
```diff
@@ -53,7 +53,11 @@ def ssh_exec(ssh_client, cmd):
 def quilt_ssh(ip):
 	client = paramiko.SSHClient()
 	client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+<<<<<<< HEAD
+	client.connect(ip, username="quilt", password="SSH_PASSWORD")
```
Comment (mchang6137) [inline src/measure_performance_MEAN_py3.py]: Why do you need SSH_PASSWORD? It should only require your private key to connect to the remote server right?

File: src/modify_resources.py [hunk 2]
```diff
@@ -82,232 +86,239 @@ def remove_all_network_manipulation(ssh_client, remove_all_machines=False):
...
-        throttled_containers = []
-        
-        for container_name in container_names:
-                if container_name not in container_blacklist:
-                        update_command = 'docker update --cpu-period {} --cpu-quota {} {}'.format(cpu_period, cpu_quota, container_name)
-                        print update_command
-                        ssh_client.exec_command(update_command)
-                        throttled_containers.append(container_name)
+    container_names = get_container_names(ssh_client)
+    # cpu_quota = int(cpu_quota * get_num_cores(ssh_client) * 1000000)
```
Comment (mchang6137) [inline src/modify_resources.py]: Remove comments.

File: src/modify_resources.py [hunk 3]
```diff
@@ -82,232 +86,239 @@ def remove_all_network_manipulation(ssh_client, remove_all_machines=False):
...
+    print 'Adjusted CPU period: {}'.format(cpu_period)
+    print 'CPU Quota: {}'.format(cpu_quota)
+
+    throttled_containers = []
+    
+    core = 0
+    for container_name in container_names:
+        if container_name not in container_blacklist:
+            update_command = 'docker update --cpu-period {} --cpu-quota {} --cpuset-cpus {} {}'.format(cpu_period, cpu_quota, core, container_name)
+            print update_command
```
Comment (mchang6137) [inline src/modify_resources.py]: Remove this print.

File: src/modify_resources.py [hunk 4]
```diff
@@ -82,232 +86,239 @@ def remove_all_network_manipulation(ssh_client, remove_all_machines=False):
...
+            ssh_client.exec_command(update_command)
+            throttled_containers.append(container_name)
+            core += 1
```
Comment (mchang6137) [inline src/modify_resources.py]: Is the assumption that each container is only pinned to a single core? Also, you should not assume that container_name[0] is assigned to the 0th core, container_name[1] is assigned to the 1st core, etc. Why update the cpuset anyways?

File: src/modify_resources.py [hunk 5]
```diff
@@ -82,232 +86,239 @@ def remove_all_network_manipulation(ssh_client, remove_all_machines=False):
...
-        
-        for container_name in container_names:
-                if container_name not in container_blacklist:
-                        update_command = 'docker update --cpu-period {} --cpu-quota {} {}'.format(cpu_period, cpu_quota, container_name)
-                        print update_command
-                        ssh_client.exec_command(update_command)
-                        throttled_containers.append(container_name)
+    container_names = get_container_names(ssh_client)
+    # cpu_quota = int(cpu_quota * get_num_cores(ssh_client) * 1000000)
+    cpu_quota = int((float(cpu_quota) / 10) * 1000000)
```
Comment (mchang6137) [inline src/modify_resources.py]: Isn't there some assumption about the CPU quota being some multiple of the number of cores in the machine?

**Review by mchang6137 — CHANGES_REQUESTED** [review]
You need some considerations about the number of cores on the machines. Look into this on the Redhat documentation. It looks like there is also some assumption about containers being on certain cores? This assumption will need to be removed. Otherwise, looks good.

**Comment by atalreja** [comment]
Fixed changes that you commented on and verified that it works without making the assumption of pinning to a core and using cpuset. The skeleton code that you had for cpu quota turned out to be the correct way to do it.

**Pull request merged**